File: 2022/day1/main.js
const fs = require('fs'),
    readline = require('readline');

const rd = readline.createInterface({
    input: fs.createReadStream('input.txt'),
    console: false
});
async function processLines() {

    let max = [0, 0, 0]
    let sum = 0

    for await (const line of rd) {
        if (line === '') {
            if (sum > max[0]) {
                max.push(sum)
                max.splice(0, 1)
                max.sort((a, b) => a - b)
            }
            sum = 0
        } else {
            sum += parseInt(line)
        }
    }

    return max
}

processLines().then(max => {
    console.log(max[2])
    console.log(max.reduce((a, b) => a + b, 0))
})

File: 2022/day1/main.py
#!/usr/bin/env python3
from functools import reduce


def max_calories(accumulator, current):
    calories_sum = reduce(lambda a, b: a + eval(b), current.splitlines(), 0)

    return accumulator if accumulator > calories_sum else calories_sum


def max_three_calories(accumulator, current):
    calories_sum = reduce(lambda a, b: a + eval(b), current.splitlines(), 0)

    if calories_sum > accumulator[0]:
        accumulator.append(calories_sum)
        accumulator.pop(0)
        accumulator.sort()

    return accumulator


with open('input.txt', 'r') as f:
    lines = tuple(f.read().split(sep='\n\n'))
    print(reduce(max_calories, lines, 0))
    print(sum(reduce(max_three_calories, lines, [0, 0, 0])))
